enhancements/ingress/ingress-operator-operand-logging-level.md

```diff
@@ -165,6 +194,44 @@ Presumably, in some cases, rolling out new router pods may "reset" the issue and
 
 It is worth noting the openshift-sdn team has already taken a similar approach for controlling the logging verbosity of some OpenShift networking components.
 
+#### Log level names in the Ingress Operator
+
+The most verbose named log level available in Zapr is debug, leaving no appropriate log level to map to "Trace" or
+"TraceAll". However, Zapr and the Zap library it's based on allow for more verbose log levels to be set using numeric
+values. Using the named log levels is reasonably straightforward:
+```go
+// This will be logged when OperatorLogLevel is set to Normal or more verbose. It will
+// show up in the logs as 'info'.
+myLogger.Info("This is an info message")
+// This will also be logged when OperatorLogLevel is Normal or more verbose. The log will
+// not differentiate between V(0) and no V() level specified, so this will also show up as
+// 'info'.
+myLogger.V(0).Info("This is an info message")
+// This will be logged when OperatorLogLevel is at least Debug. It will show up in the
+// logs as 'debug'.
+myLogger.V(1).Info("This is a debug message")
+```
+
+Trace is the next level after Debug, so it makes sense to map it to `V(2)`:
+```go
+// This will be logged when OperatorLogLevel is at least Trace. Since there is no official
+// name for Trace, it will show up as 'Level(-2)'.
+// Note that it's the inverse of the V() number specified.
+myLogger.V(2).Info("This is a trace message")
+```
+
+The logical next step is to map TraceAll to `V(3)`, but Zapr allows much higher V levels, with `V(127)` as the most
+verbose option. While it's likely that there's no reason for the Ingress Operator to use this whole range, TraceAll
+should include every log message possible, so setting `OperatorLogLevel` to TraceAll should set the internal log level
+to 127:
+```go
+// This will only be logged when OperatorLogLevel is TraceAll. It will show up as 'Level(-3)'.
+myLogger.V(3).Info("This is a traceall message")
+// This will also only be logged when OperatorLogLevel is TraceAll. It will show up as
+// 'Level(-127)'. This is the highest verbosity available.
+myLogger.V(127).Info("This is a traceall message")
+```
+
```

Comment:
Can we detail on how the log level will be set at runtime for the operator?

I suppose it's a paradigm change because for the moment the log level is [set as a global variable](https://github.com/openshift/cluster-ingress-operator/blob/e24a381634a53c3a7aabe8196faf840a6ad9a041/pkg/log/log.go#L15-L26) which then reused in the other packages.